### DISCUSSION TITLE
/wind crashes client.

### DISCUSSION THREAD

**Discussion by HiddenStriker** [Pryaxis/TShock#3017] [General]
I figured this out when trying to spawn a thunderstorm. There was no command for it specifically, so I decided to spawn it by starting rain with .worldevent rain then setting .wind to 60. However, when it ended and I wanted the thunderstorm again, I did wind 60 and my client crashed. I thought maybe I set the value too high or something, but the wiki said wind ranges from 0 to 60  mph. Regardless I tried with .wind 20, and no matter what, using .wind twice seems to crash my client.

**Comment by HiddenStriker** [comment]
Here, a screenshot of the error I get when my client crashes because of a .wind command.

https://prnt.sc/oa8BWYJ3vmN6

**Comment by punchready** [comment]
The range for the command is -0.8 to 0.8

**Reply by HiddenStriker** [reply]
https://prnt.sc/2kXxdklAlMdb

I cannot get that to work

Edit: https://prnt.sc/bzE7ZSWAraU9 Tried it in many other ways and it just gives out invalid wind speed

**Reply by punchready** [reply]
My mistake, the command attempts to let the user input the wind speed in mph but is implemented incorrectly, it would be supposed to convert it to the 0.8 to -0.8 range.

**Reply by punchready** [reply]
I've created https://github.com/Pryaxis/TShock/pull/3018 to fix it.
